pkg/controller/controller_utils.go

```diff
@@ -1127,7 +1127,9 @@ func RemoveTaintOffNode(ctx context.Context, c clientset.Interface, nodeName str
 
 // PatchNodeTaints patches node's taints.
 func PatchNodeTaints(ctx context.Context, c clientset.Interface, nodeName string, oldNode *v1.Node, newNode *v1.Node) error {
-	oldData, err := json.Marshal(oldNode)
+	oldNodeNoRV := oldNode.DeepCopy()
+	oldNodeNoRV.ResourceVersion = ""
```

Comment:
add comment for why this is necessary. Also, a shallow copy should be enough.

Reply:
Done.

Reply:
Reverted, leaving as a deep copy to make sure that we do not modify object that was passed to us. It seems not worth the performance gain.